src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/DocUpdateSentinel.java

```diff
@@ -377,7 +370,9 @@ private boolean doSaveImpl(final String path,
       // Don't auto-save when there are no changes. In addition to being
       // wasteful, it causes the server to think the document is dirty.
       if (path == null && fileType == null && diff.isEmpty()
-          && foldSpec.equals(oldFoldSpec) 
+          && newFoldSpec.equals(oldFoldSpec)
+          && newSelectionSpec.equals(oldSelectionSpec)
```

Comment:
If it turns out to be too heavyweight to trigger autosaves as the cursor moves, we could try triggering a cursor position save only when the page unloads and/or the tab loses focus -- this would introduce a couple of pathological situations (e.g. you'd lose position after a crash) but would also dramatically reduce the overhead.